.clj-kondo/config.edn

```diff
@@ -658,6 +658,8 @@
    :name    printable-namespaces}
   {:pattern "^metabase\\.lib\\.*"
    :name    metabase-lib}
+  {:pattern "^metabase\\.(?!(api|query-processor|server)).*$"
+   :name    non-api-namespaces}
```

Comment:
I think you should be explicit about calling this the REST API rather than just the API, because drivers for example have an API, but that's not what we're talking about here

```suggestion
   :name    non-rest-api-namespaces}
```